### ISSUE TITLE
v2.9.1: The conditional check 'k3s_check_cgroup_option.rc == 0' failed. The error was: error while evaluating conditional (k3s_check_cgroup_option.rc == 0): 'dict object' has no attribute 'rc'

### ISSUE THREAD

**Issue by cawolf** [PyratLabs/ansible-role-k3s#116]
### Summary

On version 2.9.1, the task `xanmanning.k3s : Fail if memory cgroup is not enabled` fails with `'dict object' has no attribute 'rc'`. This issue first seemed to be a regression of #111 , but I experienced the same error on 2.8.3 and up. I also downgraded down to 2.7.0 and the error persisted, so maybe there is a user error here?

### Issue Type

- Bug Report

### Controller Environment and Configuration

role version: v2.9.1
host: Ubuntu 20.04.2 (64 bit)
targets: Debian GNU/Linux 10 (64 bit) on Raspberry Pi 4

### Steps to Reproduce

execute this playbook:
```yaml
- name: Build a cluster with a single control node
  hosts: k3s_cluster
  vars:
    k3s_become_for_all: true
    k3s_server:
      disable:
        - traefik
        - servicelb
        - local-storage
    k3_agent:
      kubelet-arg: "--authentication-token-webhook=true --authorization-mode=Webhook"
  roles:
    - xanmanning.k3s
```

### Expected Result

The playbook runs and succeeds as usual. It worked in versions 2.2.0 and 

### Actual Result

```text
TASK [xanmanning.k3s : Check if memory cgroup is enabled] ******************************************************************************************************************************************************************************
ok: [k3s-2]
ok: [k3s-1]

TASK [xanmanning.k3s : Fail if memory cgroup is not enabled] ***************************************************************************************************************************************************************************
fatal: [k3s-1]: FAILED! => {"msg": "The conditional check 'k3s_check_cgroup_option.rc == 0' failed. The error was: error while evaluating conditional (k3s_check_cgroup_option.rc == 0): 'dict object' has no attribute 'rc'"}
fatal: [k3s-2]: FAILED! => {"msg": "The conditional check 'k3s_check_cgroup_option.rc == 0' failed. The error was: error while evaluating conditional (k3s_check_cgroup_option.rc == 0): 'dict object' has no attribute 'rc'"}

```

**Comment by xanmanning** [comment]
Hi.

Not sure how to recreate this. What version of Ansible are you running, and on what version of Python?

The reason why I ask is the command module ALWAYS returns an `.rc` (see [ansible.builtin.command](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/command_module.html#return-rc)) and the command will always run (even in [check mode](https://github.com/PyratLabs/ansible-role-k3s/blob/main/tasks/validate/environment/remote/cgroups.yml#L8) which was fixed in #111)

**Comment by cawolf** [comment]
Hey,
this is the output on my contrlling machine:
```
ansible 2.10.3
  config file = None
  configured module search path = ['/home/cwolf/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/cwolf/.local/lib/python3.8/site-packages/ansible
  executable location = /home/cwolf/.local/bin/ansible
  python version = 3.8.5 (default, Jan 27 2021, 15:41:15) [GCC 9.3.0]
```
```
ansible-playbook 2.10.3
  config file = None
  configured module search path = ['/home/cwolf/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/cwolf/.local/lib/python3.8/site-packages/ansible
  executable location = /home/cwolf/.local/bin/ansible-playbook
  python version = 3.8.5 (default, Jan 27 2021, 15:41:15) [GCC 9.3.0]
```

It worked when I initially set up the ansible playbook (role was on 2.2.0), but no other role version is working for me. Do you know of any role cache or python cache I could get rid of?

Or do you have a very basic playbook that I could test locally?

**Comment by sergeyshaykhullin** [comment]
@xanmanning  I am facing with the same problem

```
TASK [xanmanning.k3s : Fail if memory cgroup is not enabled] *******************
fatal: [node1]: FAILED! => {"msg": "The conditional check 'k3s_check_cgroup_option.rc == 0' failed. The error was: error while evaluating conditional (k3s_check_cgroup_option.rc == 0): 'dict object' has no attribute 'rc'"}
fatal: [node2]: FAILED! => {"msg": "The conditional check 'k3s_check_cgroup_option.rc == 0' failed. The error was: error while evaluating conditional (k3s_check_cgroup_option.rc == 0): 'dict object' has no attribute 'rc'"}
fatal: [node3]: FAILED! => {"msg": "The conditional check 'k3s_check_cgroup_option.rc == 0' failed. The error was: error while evaluating conditional (k3s_check_cgroup_option.rc == 0): 'dict object' has no attribute 'rc'"}
fatal: [node4]: FAILED! => {"msg": "The conditional check 'k3s_check_cgroup_option.rc == 0' failed. The error was: error while evaluating conditional (k3s_check_cgroup_option.rc == 0): 'dict object' has no attribute 'rc'"}
```

Debug output
```
ok: [node1] => {
    "k3s_check_cgroup_option": {
        "changed": false,
        "failed": false,
        "failed_when_result": false,
        "msg": "Unsupported parameters for (ansible.legacy.command) module: cmd Supported parameters include: _raw_params, _uses_shell, argv, chdir, creates, executable, removes, stdin, stdin_add_newline, strip_empty_ends, warn"
    }
}
ok: [node2] => {
    "k3s_check_cgroup_option": {
        "changed": false,
        "failed": false,
        "failed_when_result": false,
        "msg": "Unsupported parameters for (ansible.legacy.command) module: cmd Supported parameters include: _raw_params, _uses_shell, argv, chdir, creates, executable, removes, stdin, stdin_add_newline, strip_empty_ends, warn"
    }
}
ok: [node3] => {
    "k3s_check_cgroup_option": {
        "changed": false,
        "failed": false,
        "failed_when_result": false,
        "msg": "Unsupported parameters for (ansible.legacy.command) module: cmd Supported parameters include: _raw_params, _uses_shell, argv, chdir, creates, executable, removes, stdin, stdin_add_newline, strip_empty_ends, warn"
    }
}
ok: [node4] => {
    "k3s_check_cgroup_option": {
        "changed": false,
        "failed": false,
        "failed_when_result": false,
        "msg": "Unsupported parameters for (ansible.legacy.command) module: cmd Supported parameters include: _raw_params, _uses_shell, argv, chdir, creates, executable, removes, stdin, stdin_add_newline, strip_empty_ends, warn"
    }
}
```

**Comment by sergeyshaykhullin** [comment]
I found that issue in 2.10.3, but now i can't anything about it =(

https://github.com/arillso/docker.ansible/pull/48

Does this role can handle this error somehow?

**Comment by xanmanning** [comment]
I was asked to change the call to the command module to add 2.9.16 support: https://github.com/PyratLabs/ansible-role-k3s/issues/105#issuecomment-817182233

The role requirements were updated to reflect this: https://github.com/PyratLabs/ansible-role-k3s#requirements

Unfortunately the move to FQCN means I am targeting `ansible.builtin.command` - however it looks like (from your output) `2.10.4` is calling `ansible.legacy.command` which doesn't have the command argument.

In a nutshell to avoid occasionally banjaxed calls to modules and collections in Ansible (as a whole):

For the **2.9** release, you need >= 2.9.16
For the **2.10** release, you need >= 2.10.4

**Comment by cawolf** [comment]
This solved it for me. Upgrading ansible via `pip` installed versions 2.10.10 (ansible-base) and 2.10.7 (ansible) on my machine. With this versions and the role version 2.9.1, everything works like a charm. Thank you for your time :)
